### DISCUSSION TITLE
[Missing Feature] Is there a way to setup login with OTP?

### DISCUSSION THREAD

**Discussion by manojLondhe** [Sunbird-Lern/Community#133] [Query] [answered]
Hi,

Wanted to know if there is a way to setup login with OTP by changing keycloak setup?

If not, is there a plan for same to be added?

**Comment by Jayaprakash8887** [comment]
@manojLondhe At the moment, we have not enabled this flow in Sunbird and there is no plan for this feature in the pipeline. Would appreciate if you could do the necessary development and contribute to Sunbird LERN. Looking forward to it. Thank you.

**Reply by manojLondhe** [reply]
Thank you for answering.

Could we get top level steps that might be required, so we can give it try?

**Reply by Jayaprakash8887** [reply, marked as answer]
https://lern.sunbird.org/learn/product-and-developer-guide/user-and-org-service/keycloak-admin-configuration

**Comment by ankitv89** [comment]
Hi @manojLondhe ,

We have enabled this at  Aastrika [sphere.aastrika.org]  by modying the theme and SPI. You can checkout the code here: https://github.com/Sphere/sunbird-auth/tree/feature/updatePasswordTheme

Changes : sms-provider/keycloak and in themes.
